src/Database/LSMTree/Internal/Serialise/Class.hs

```diff
@@ -1,43 +1,70 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+
 -- | Public API for serialisation of keys, blobs and values
 module Database.LSMTree.Internal.Serialise.Class (
     SerialiseKey (..)
+  , serialiseKeyIdentity
+  , serialiseKeyPreservesOrdering
+  , serialiseKeyMinimalSize
   , SerialiseValue (..)
+  , serialiseValueIdentity
+  , serialiseValueConcatDistributes
   , RawBytes (..)
   ) where
 
-import           Database.LSMTree.Internal.Serialise.RawBytes
+import           Data.Proxy (Proxy)
+import           Database.LSMTree.Internal.Serialise.RawBytes as RawBytes
 
--- | Ordering-preserving serialisation for keys.
+-- | Serialisation of keys.
+--
+-- Instances should satisfy the following:
+--
+-- [Identity] @'deserialiseKey' ('serialiseKey' x) == x@
+--
+-- === CompactIndex constraints
+--
+-- When using the 'CompactIndex', additional constraints apply to the
+-- serialisation function, so in that case instances should also satisfy the
+-- following:
+--
+-- [Ordering-preserving] @x \`'compare'\` y == 'serialiseKey' x \`'compare'\` 'serialiseKey' y@.
```

Comment:
I think that one would be reasonable to require for all types of index, but right, doesn't necessarily have to.

Reply:
You're right, we're also depending on this in other places -- I'm going to move it outside of the constraints that are specific to the compact index